src/Runner/Runner.php

```diff
@@ -460,10 +465,46 @@ private function fixFile(\SplFileInfo $file, LintingResultInterface $lintingResu
         $new = $old;
         $newHash = $oldHash;
 
+        $ruleCustomisers = null === $this->ruleCustomisationPolicy ? [] : $this->ruleCustomisationPolicy->getRuleCustomisers();
+        if ([] !== $ruleCustomisers) {
+            $usedFixerNames = array_map(
+                static fn (FixerInterface $fixer): string => $fixer->getName(),
+                $this->fixers
+            );
+            $missingFixerNames = array_diff(array_keys($ruleCustomisers), $usedFixerNames);
+            if ([] !== $missingFixerNames) {
+                $missingFixerNames = implode("\n- ", $missingFixerNames);
+
+                throw new \RuntimeException(
+                    <<<EOT
+                        Rule Customisation Policy contains customisers for fixers that are not currently applied.
+
+                        Missing fixers:
+                        - {$missingFixerNames}
+
+                        Please check your configuration to ensure that these fixers are included, and that they have not been replaced by other fixers in the version of PHP-CS-Fixer you are using.
```

Comment:
*major*

we shall elaborate on deprecation case.

if set includes `visibility_required`, Customization Policy has a lambda for it, and then rule got deprecated and replaced in set by `modifier_keywords`,
how we handle it here? simply with this exception? then, we need to tell the user about the reason of the problem. User did not replaced anything about `visibility_required` from his perspective, he may not know it got deprecated and rule he was using was updated.

also, such crash after upgrade will be consider BC break, imho - i prefer we handle it differently.